src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -649,6 +649,19 @@ export default class DefaultDeviceController
   }
 
   setDeviceLabelTrigger(trigger: () => Promise<MediaStream>): void {
+    if (this.deviceInfoCache) {
+      let hasDeviceLabels = true;
+      for (const device of this.deviceInfoCache) {
+        if (!device.label) {
+          hasDeviceLabels = false;
+          break;
+        }
+      }
+      if (!hasDeviceLabels) {
+        this.deviceInfoCache = [];
+      }
+    }
+
```

Comment:
You should set to `null` instead, and this can be shortened:

```typescript
// Discard the cache if it was populated with unlabeled devices.
if (this.deviceInfoCache) {
  for (const device of this.deviceInfoCache) {
    if (!device.label) {
      this.deviceInfoCache = null;
      break;
    }
  }
}
```

Reply:
Addressed.